rest/utilities_testing_blip_client.go

```diff
@@ -1436,8 +1444,17 @@ func (btc *BlipTesterCollectionClient) StartPullSince(options BlipTesterPullOpti
 }
 
 func (btc *BlipTesterCollectionClient) StopPush() {
-	require.True(btc.TB(), btc.pushRunning.CASRetry(true, false), "can't stop push replication - not running")
+	require.True(btc.TB(), btc.pushRunning.IsTrue(), "can't stop push replication - not running")
 	btc.pushCtxCancel()
+
+	// Wake up any waiting push loops to check for cancellation
+	btc._seqCond.Broadcast()
+
+	// wait for push replication to stop running
+	require.EventuallyWithT(btc.TB(), func(c *assert.CollectT) {
+		require.True(c, btc.pushRunning.IsTrue() == false)
```

Comment:
```suggestion
		assert.False(c, btc.pushRunning.IsTrue())
```

the way that assertion is written will actually fail immediately if `btc.pushRunning.IsTrue() == true` and it won't do a backoff.